### DISCUSSION TITLE
Trying to use C/C++ for Open Enterprise Languages on z/OS to port

### DISCUSSION THREAD

**Discussion by fswarbrick** [zopencommunity/meta#753] [Porting]
Getting the following errors:
```
VERBOSE: Setting up /u/dvfjs/zopen/usr/local/zopen/comp_xlclang/comp_xlclang dependency environment
FSUM3224 xlclang: Fatal error in /usr/lpp/cbclib/xlclang/exe/clcdrvr: signal 9 received.
xlclang compiler specified, but it is not a 'IBM C/C++ for Open Enterprise Languages on z/OS' compiler


***ERROR: Configure (full) failed. Log: /u/dvfjs/jqport/log.DEV/20240417_182711_config.log
./configure: eval: runAndLog 3: configure 17: /u/dvfjs/zopen/usr/local/bin/zopen-build 2287: FSUM7351 not found
kill: cleanupOnExit 7: exit: printError 7: configure 21: /u/dvfjs/zopen/usr/local/bin/zopen-build 2287: 131856: EDC5139I Operation not permitted.
zopen-build completed in 17 seconds.
``` 
Open Enterprise Languages supplies clang, clang++, clang-cl and clang-cpp.  How do I get the build to use clang instead of xlclang?  Or am I perhaps doing something else wrong?

**Comment by v1gnesh** [comment]
Have you had a look at https://github.com/ZOSOpenTools/jqport/blob/main/buildenv ?

**Comment by IgorTodorovskiIBM** [comment]
It seems your xlclang may not be set it properly:
```
xlclang: Fatal error in /usr/lpp/cbclib/xlclang/exe/clcdrvr: signal 9 received.
```
Usually this indicates that the datasets are not where they should be. `clcdrvr` is an external link to a dataset member containing the main compiler driver program.

You can choose the clang compiler with `export ZOPEN_COMP=CLANG`.

Which project are you trying to build? If there's no configure script, then you can skip that with:
```
export ZOPEN_CONFIGURE=skip
```

**Reply by IgorTodorovskiIBM** [reply]
Are you going through these instructions? https://zosopentools.org/#/Guides/Porting

**Reply by fswarbrick** [reply]
I'm trying to follow the example for jq.  With setting ZOPEN_COMP=CLANG (really, using that entire buildenv file you linked to) it now compiles successfully.  Unfortunately, it crashes upon use.

```shell
echo '{"foo": 0}' | jq .

error: cannot allocate memory
 1: 0x1061ED7E backtrace_symbols_fd+0x94 [zopen-ci3:/jenkins/workspace/Port-Build/zoslib/zoslib/src/zos.cc:449]
 2: 0x1061EEC0 __display_backtrace+0x88 [zopen-ci3:/jenkins/workspace/Port-Build/zoslib/zoslib/src/zos.cc:456]
 3: 0x105815AC abort+0x92 [zopen-ci3:/jenkins/workspace/Port-Build/zoslib/zoslib/src/zos.cc:705]
 4: 0x10560A40 jv_mem_calloc+0x80
 5: 0x1051F278 block_compile+0x5f4
 6: 0x1050D9FA jq_compile_args+0x434
 7: 0x1070AADE main+0x175e
 8: 0x1070AEFA CELQINIT+0x1aca [CELQINIT:]
CEE5207E The signal SIGABRT was received.
SIGABRT
```

**Comment by fswarbrick** [comment]
I got it working.  I had to fix several issues:

- The tutorial does not refer to an additional patch needed, the "zOS.patch" (https://github.com/ZOSOpenTools/jqport/blob/92efb81a9ce68b030d39eba3376b54928f86e299/patches/zOS.patch).
- The .Plo files placed in jqport/jq-1.6/src/.deps were all tagged as ISO-8859-1 (ASCII), but were in fact EBCDIC 1047.  Except for inject_errors.Plo which was, in fact, ASCII.  So I chtag'd all of the others to IBM-1047.

I don't think there were any other issues, but I might have forgotten.
